cpp/src/arrow/flight/sql/example/sqlite_server.cc

```diff
@@ -305,7 +305,7 @@ class SQLiteFlightSqlServer::Impl {
                           EncodeTransactionQuery(query, command.transaction_id));
     std::vector<FlightEndpoint> endpoints{FlightEndpoint{std::move(ticket), {}}};
     ARROW_ASSIGN_OR_RAISE(auto result,
-                          FlightInfo::Make(*schema, descriptor, endpoints, -1, -1))
+                          FlightInfo::Make(*schema, descriptor, endpoints, -1, -1, false))
```

Comment:
SQLite doesn't tell us whether the result set is sorted or not, so I suppose technically this should always be 'true'? (That said, this server also only ever returns one endpoint so it's moot either way.)

Reply:
We can use `true` here because this flight server always returns one endpoint.
But I think that `false` is better here because `false` means "unknown order".

A real Flight SQL server will set `true` here only when it analyzes the given query and detects `ORDER BY` in a main `SELECT`.
How about adding a comment something like "TODO: Set ordered to true only when the query has ORDER BY" here?